src/textual/widgets/_selection_list.py

```diff
@@ -646,6 +646,11 @@ def _remove_option(self, index: int) -> None:
         option = self.get_option_at_index(index)
         self._deselect(option.value)
         del self._values[option.value]
+        # Decrement index of options after the one we just removed.
```

Comment:
Maybe a separate issue (or not an issue at all if I'm missing something), but this makes me wonder why the `highlighted` index wouldn't be decremented if it appears below the removed options.

Reply:
Actually from a skim, I'm guessing it might be that the highlight is applied based on the ID of the option rather than the index. It's just converted to an ID via a validator.